internal/controller/helmrelease_controller.go

```diff
@@ -147,6 +156,10 @@ func (r *HelmReleaseReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if !isValidChartRef(obj) {
+		return ctrl.Result{}, reconcile.TerminalError(fmt.Errorf("invalid Chart reference"))
+	}
```

Comment:
We use CEL validation to have a `oneOf` validation behaviour. Quoting here "Kubernetes CRD Validation Using CEL has been turned on by default since Kubernetes 1.25 and graduated to GA in Kubernetes 1.29." As of now we support k8s from 1.27 so most likely it is safe, but I assume that it is still a possible scenario.